src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,47 +1,83 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+interface TodoModalProps {
+  setIsOpenModal: React.Dispatch<React.SetStateAction<boolean>>,
+  activeTodo: Todo | null,
+  setActiveTodo: React.Dispatch<React.SetStateAction<Todo | null>>
+}
+
+export const TodoModal: React.FC<TodoModalProps> = (props: TodoModalProps) => {
+  const { setIsOpenModal, activeTodo, setActiveTodo } = props;
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const closeModal = () => {
+    setIsOpenModal(false);
+    setActiveTodo(null);
+  };
+
+  useEffect(() => {
+    if (!activeTodo) {
+      return;
+    }
+
+    getUser(activeTodo.userId).then((data) => {
+      setUser(data);
```

Comment:
Consider handling errors in the 'getUser' promise using a catch block to prevent unhandled rejections.